.github/workflows/doc-tests.yaml

```
@@ -87,8 +76,17 @@ jobs:
       - name: Check spelling
         run: cd $GITHUB_WORKSPACE/docs && yarn spellcheck content/teleport
 
-      - name: Lint the docs
-        run: cd $GITHUB_WORKSPACE/docs && yarn markdown-lint
+      - name: Lint Markdown in changed files
+        run: |
+          cd $GITHUB_WORKSPACE/docs
+          npx remark --rc-path .remarkrc.mjs --frail --ignore-pattern '**/includes/**' --silently-ignore ${{ needs.changes.outputs.changed_files }}
+
+      - name: Lint style in changed files
+        uses: errata-ai/vale-action@reviewdog 
```

Can we pin to a hash here instead of using the `reviewdog` tag.  I'd recommend their most recent release `v2.0.1`

```suggestion
        uses: errata-ai/vale-action@c99f2dfd2aeaedb3d4bb16f385841830b9164d31 # reviewdog
```

Lets also pin the vale version ([docs](https://github.com/errata-ai/vale-action#version-default-latest)).  2.30.0 seems to be the most current.

We do this so that the action doesn't start failing and blocking PRs due to changes irrelevant to the PR at hand.  E.g. see the following for this mode of failure:

* https://gravitational.slack.com/archives/C05NU0A2Q8P/p1697819972564169
* https://gravitational.slack.com/archives/C0DF0TPMY/p1693347947400259

The risk here is that dependabot isn't smart enough to update the vale version (afaik).  It will only keep the action sha up to date.

Also, do you know what is up with the `reviewdog` branch?  This action is highly unusual in that it doesn't put out regularly tagged semver releases (`v2.0.1` was a year and a half ago).  Instead it has a wierdly named "main" branch that it recommends folks use.  It doesn't inspire confidence that this team follows conventional GitHub actions patterns.

---

Thanks! And yeah, I'm not sure what's going on with `reviewdog`—I blindly followed the usage instructions here!